### PR TITLE
Allow Reference Validator to reference specific Assets

### PR DESCRIPTION
**Pull request by Chownas**
Reference Validator now also accepts direct links to file. Also did some cleanup in the project.

### CODE REVIEW

File: Plugins/OWI_EditorEnhancements/Source/OWI_EditorEnhancements/Private/OWI_ReferenceRulesHandler.cpp [hunk 1]
```diff
@@ -8,11 +8,25 @@ bool FOWIReferenceRule::IsViolated(const FString& AssetPath, const TArray<FName>
...
+			continue;
+		}
+
 		if (Reference.ToString().StartsWith(MayNotReferenceFolder.Path)
-			&& AssetPath.StartsWith(MayNotReferenceFolder.Path) == false
-			&& IsException(Reference) == false)
+			&& AssetPath.StartsWith(MayNotReferenceFolder.Path) == false)
 		{
-			ValidationError = AssetPath + " references " + Reference.ToString() + " in invalid folder " + MayNotReferenceFolder.Path;
+			ValidationError = AssetPath + " references " + Reference.ToString() + " in folder " + MayNotReferenceFolder.Path;
```
Comment (Mat-Stevenson-OWI) [inline Plugins/OWI_EditorEnhancements/Source/OWI_EditorEnhancements/Private/OWI_ReferenceRulesHandler.cpp]: I d rather see a sprintf/printf here but no biggy

File: Plugins/OWI_EditorEnhancements/Source/OWI_EditorEnhancements/Private/OWI_ReferenceRulesHandler.cpp [hunk 2]
```diff
@@ -22,7 +36,19 @@ bool FOWIReferenceRule::IsViolated(const FString& AssetPath, const TArray<FName>
 
 bool FOWIReferenceRule::CanBeApplied(const FString& AssetPath) const
 {
-	return AssetPath.StartsWith(AssetFolder.Path);
+	const FString AssetFilePath = ConvertFilePath(AssetFile);
+	return AssetPath.StartsWith(AssetFolder.Path) || AssetPath.Equals(AssetFilePath);
+}
+
+FString FOWIReferenceRule::ConvertFilePath(const FFilePath FilePath)
```
Comment (Mat-Stevenson-OWI) [inline Plugins/OWI_EditorEnhancements/Source/OWI_EditorEnhancements/Private/OWI_ReferenceRulesHandler.cpp]: Can FilePath be passed by reference?

File: Plugins/OWI_EditorEnhancements/Source/OWI_EditorEnhancements/Private/OWI_ReferenceRulesHandler.cpp [hunk 3]
```diff
@@ -144,12 +179,23 @@ bool UOWIReferenceRulesHandler::IsBlackListed(const FString& AssetPath, const TA
...
 	for (const FName& Reference : References)
 	{
-		for (const FDirectoryPath& BlackListItem : Settings->BlackList)
+		for (const FDirectoryPath& BlackListItem : Settings->BlackListFolders)
 		{
 			if (Reference.ToString().StartsWith(BlackListItem.Path)
 				&& AssetPath.StartsWith(BlackListItem.Path) == false)
 			{
-				ValidationError = AssetPath + " references " + Reference.ToString() + " in invalid folder " + BlackListItem.Path;
+				ValidationError = AssetPath + " references " + Reference.ToString() + " in folder " + BlackListItem.Path;
```
Comment (Mat-Stevenson-OWI) [inline Plugins/OWI_EditorEnhancements/Source/OWI_EditorEnhancements/Private/OWI_ReferenceRulesHandler.cpp]: Maybe sprintf/printf

**Review by Mat-Stevenson-OWI — APPROVED** [review]
Nice job.

**Review by regner — APPROVED** [review]
In the future can we please have changes such as the copyright notice done in their own commit. Preferably their own PR as well, but at least their own commit.

**Pull request merged**